doc/source/index.rst

```diff
@@ -73,7 +73,6 @@ Wherever possible, do not take these conversations to private channels, includin
    examples/quickstart
    faq
    examples/README
-   API docs <https://docs.mdanalysis.org/stable/index.html>
```

Comment:
Does development docs below also want to be there?

Reply:
Thanks, good pickup!